main.cpp

```diff
@@ -1,25 +1,57 @@
 #include <iostream>
+#include <getopt.h>
 #include "common.h"
-#include "md5-duplicate-detection.h"
 #include "encrypt-decrypt.h"
 #include "files.h"
 
-int main() {
-    std::cout << "Hello, World!" << std::endl;
-    /*
-    std::string directory = "some directory";
+int main(int argc, char *argv[]) {
+
+    int opt;
+    const char *directory;
     std::vector<fs::path> files;
-    getFiles(directory, files);
-    for (fs::path i: files)
-        std::cout << i.string() << ' ';
-        */
 
-//    fs::path A("test-files/A");
-//    fs::path B("test-files/B");
-//    is_data_duplicate(A,B);
+    while (true) { //getopt creates a data-structure that can be looped through from the argv array to process the parameters and their arguments
+
+        int option_index = 0;
+        static struct option long_options[] =
+                {
+                        {"help",     no_argument, NULL,  'h' },
+                        {"encrypt",  no_argument, NULL,  'e' },
+                        {"decrypt",  no_argument, NULL,  'd' },
+                        {NULL,      0,    NULL,   0  }
+                };
+        opt = getopt_long(argc, argv, "he:d:", long_options, &option_index);
+
+        if(opt == -1)
+            break;
+
+        switch (opt) {
+            case 'h':
+                printf("ex1 -e --encrypt -d --decrypt\n"
+                       "-e --encrypt \".../path-to-root\"  | Encrypts files in a given directory and accepts the root path as an argument\n"
+                       "-d --decrypt \".../path-to-root\"  | Decrypts encrypted files from this program and accepts the root path as an argument\n");    //placeholder
+                break;
+            case 'e':
+                directory = optarg;
+                getFiles(directory, files);
+                for (fs::path &i: files) {
+                    encrypt(i);
+                    std::cout << "Encrypted " << i.string() << "\n";
```

Comment:
You can insert the `fs::path` directly to the `std::cout` stream, you don't need to first get the string of it.

This is also redundant as the encrypt command already prints a line. You should change that code if you want to print more information or change the phrasing. Which should probably happen. Something about `Encryption complete. Output can be found in "A"` strikes me as odd.

Reply:
That's not my code mine just prints "Encrypted ./filepath..." but I'll look into it